include/rocksdb/table.h

```diff
@@ -107,6 +115,12 @@ struct BlockBasedTableOptions {
   // point to a nullptr object.
   bool no_block_cache = false;
 
+  // If non-nullptr will use this allocator instead of system allocator when
+  // allocating memory for cache blocks.
+  // You should hold an instance of the the cache_allocator alive as long
```

Comment:
I guess I don't understand the scope of `BlockBasedTableOptions`. Are those only used at construction and never copied inside RocksDB? Then you should hold a vector (you need to remove from it when DBs that use the cache are closed).

Can the allocator be a property of the block cache rather than a property of each individual DB that shares the cache? It seems strange to have blocks allocated with different allocators in the same cache.